Dockerfile.dev

```diff
@@ -17,6 +17,7 @@ RUN \
         libcairo2 \
         gdb \
         git \
+        build-essential \
```

Comment:
I can confirm that using "docker compose build --no-cache" on a Pi 3 and Pi 4 only finishes successfully when adding the propoesed dependency. Otherwise, building psutil fails.
I support the propoesed pull request.
Please consider merging. Thanks.

Reply:
I've noticed the same in the add-on. Those are only required during pip install, so ideally we'd also remove them again.

FWIW, for the add-on I've only installed the required three packages (see [here](https://github.com/home-assistant/addons/blob/master/matter_server/Dockerfile#L89-L91)). `build-essential` probably brings those and much more.